kartothek/io/dask/repartition.py

```diff
@@ -0,0 +1,92 @@
+import math
+from typing import Optional
+
+import dask
+import dask.dataframe as dd
+
+import kartothek.io.dask.dataframe as ktk_ddf
+from kartothek.core.factory import DatasetFactory
+from kartothek.serialization import PredicatesType
+
+
+def calculate_repartition_ratio(
+    dataset_factory: DatasetFactory,
+    table_name: str,
+    predicates: Optional[PredicatesType] = None,
+    *,
+    target_nrows_per_partition: int,
+    dataset_metadata_frac: float,
+) -> dd.core.Scalar:
+    """
+    Calculate a `repartition_ratio` according to the dataset input and `target_nrows_per_partition` (the desired
+    number of rows of each dataframe partition).
+
+    Parameters
+    ----------
+    dataset_factory
+        dataset factory
+    table_name
+        Dataset Table name
+    predicates
+        Optional , but predicates to be applied during calculatin repartition ratio
+    target_nrows_per_partition
+        target num of rows per partition
+    dataset_metadata_frac
+        used during collection of dataset metadata
+    Returns
+    -------
+    repartition_ratio
+        calculated repartition_ratio
+    """
+    dataset_metadata = ktk_ddf.collect_dataset_metadata(
+        factory=dataset_factory,
+        predicates=predicates,
+        table_name=table_name,
+        frac=dataset_metadata_frac,
+    )
+    # Once the kartothek issue is resolved, one should use the rowgroup wise
+    # metadata. This would set up future improvements once kartothek is able to
+    # filter RG predicates while collecting statistics (assuming this will ever
+    # be possible this way)
+    # See https://github.com/JDASoftwareGroup/kartothek/pull/352
```

Comment:
The PR linked here is merged. Can we make the change suggested in the comment?

Reply:
@lr4d ,
What is the RG attribute I should use here . 
We used `number_rows_total` till now. 
Should I use `number_row_groups`* `number_rows_per_row_group` to get total number of rows?

Reply:
From the phrasing it sounds like just `number_rows_per_row_group` but I don't see how that applies here as we are loading each Parquet file as a partition, not each row-group. @fjetter mind elaborating a bit about what you meant here?

Reply:
The original implementation was

```
expected_nrows_per_partition = dataset_metadata.groupby("partition_label")[
        "number_rows_per_row_group"
    ].sum()
```

If the collect metadata would implement predicate pushdown, this would give more accurate metrics. For now it isn't really relevant

Reply:
> If the collect metadata would implement predicate pushdown

hm, why didn't we support this originally? This would be very useful for repartitioning after reading part of a dataset, perhaps we should pick it up again

Reply:
> hm, why didn't we support this originally? This would be very useful for repartitioning after reading part of a dataset, perhaps we should pick it up again

Start small. Implement the easy feature first. Improvements and performance optimizations later. Feel free to implement the predicate pushdown properly for the stats collection but this was not included in the first iteration since it was not necessary to enable the feature. Always start with small increments...